### PR TITLE
fix: add timeout when polling the latest ask

### LINKED ISSUE #282: Docker容器后台运行会导致服务挂掉
## 描述

使用Docker容器后台detach运行，会导致服务挂掉，使用docker ps -a查询发现服务已停止。

我使用的是docker后台交互运行方式，防止因容器无前台进程导致自动停止。

## 运行命令

docker run -e OPENAI_API_KEY=xxxx -dit yihong0618/xiaogpt --restart=always --account=xxxx --password=xxxx --hardware=X10A --use_chatgpt_api --mute_xiaoai --use_command /bin/bash

### PR DESCRIPTION
**Pull request by huwei901108**
Fixes #282
the session.get process without a timeout in poll_latest_ask task may hang.
add a time out of 15sec in total.
no hang found for 24 hour run.

### CODE REVIEW

**Comment by yihong0618** [comment]
@huwei901108 

can you add some info in the content that fix: # issue_number 
and then fix the ci lint

**Comment by huwei901108** [comment]
OK.
I have changed the commit info. As well as check and reformat my code.

File: xiaogpt/xiaogpt.py [hunk 1]
```diff
@@ -229,12 +238,20 @@ def _change_prompt(self, new_prompt):
     async def get_latest_ask_from_xiaoai(self, session):
         retries = 2
         for _ in range(retries):
-            r = await session.get(
-                LATEST_ASK_API.format(
-                    hardware=self.config.hardware,
-                    timestamp=str(int(time.time() * 1000)),
+            try:
+                timeout = ClientTimeout(total = 15)
```
Comment (frostming) [inline xiaogpt/xiaogpt.py:242]: it seems unused?

Comment (huwei901108) [reply xiaogpt/xiaogpt.py:242]: Sorry. It's a typo when fixing formatting. Now timeout is added to "get()" as a parameter.

Comment (huwei901108) [reply xiaogpt/xiaogpt.py:242]: Sorry. It's a typo when fixing formatting. Now timeout is added to "get()" as a parameter.

File: xiaogpt/xiaogpt.py [hunk 2]
```diff
@@ -94,14 +94,23 @@ async def poll_latest_ask(self):
             session._cookie_jar = self.cookie_jar
             while True:
                 self.log.debug(
-                    "Now listening xiaoai new message timestamp: %s",
-                    self.last_timestamp,
+                    "listening new message, timestamp: %s",
+                    self.last_timestamp
```
Comment (yihong0618) [inline xiaogpt/xiaogpt.py:98]: f-string seems ok here
e.g. f"listening new message, timestamp: {self.last_timestamp}"

File: xiaogpt/xiaogpt.py [hunk 3]
```diff
@@ -229,12 +238,21 @@ def _change_prompt(self, new_prompt):
...
+                        timestamp = str(int(time.time() * 1000)),
+                    ),
+                    timeout = timeout
                 )
-            )
+            except Exception as e:
+                self.log.warning(
+                    "get latest ask from xiaoai meet execption: %s",
+                    str(e)
+                )
```
Comment (yihong0618) [inline xiaogpt/xiaogpt.py:254]: ditto

File: xiaogpt/xiaogpt.py (hunk 2 above)
Comment (frostming) [reply xiaogpt/xiaogpt.py:98]: no, the current form is preferred, it's not a string formatting, its lazy interpolation

Comment (yihong0618) [reply xiaogpt/xiaogpt.py:98]: copy that @huwei901108 forget me...

**Comment by yihong0618** [comment]
@huwei901108 
```
pip install -U black
black .
```

**Review by yihong0618 — APPROVED** [review]
LGTM @frostming what do you think?

**Pull request merged**